### PR TITLE
Base the Modal Size and position off the root window view

### LINKED ISSUE #6532: [bug][android]ModelPage will not work properly
### Description

When I turn on gesture navigation and turn off the gesture bar, ModelPage will not work properly

### Steps to Reproduce

[](https://user-images.githubusercontent.com/42995476/165475046-517781df-27f1-49d7-bca2-675bff1e398c.mp4)

code source:
[MauiApp3.zip](https://github.com/dotnet/maui/files/8570296/MauiApp3.zip)

### Version with bug

Release Candidate 2

### Last version that worked well

Unknown/Other

### Affected platforms

Android

### Affected platform versions

android 12

### Did you find any workaround?

_No response_

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by PureWeen**
### Description of Change

There are a [lot](https://stackoverflow.com/questions/20264268/how-do-i-get-the-height-and-width-of-the-android-navigation-bar-programmatically/50775459#50775459) of factors that go into figuring out your visible window space. These also change as you move around APIs. 

This fix just uses the positioning of the main window view to figure out where the modal should be located.

### Issues Fixed

Fixes #6532

### CODE REVIEW

File: src/Core/src/Platform/Android/ViewExtensions.cs [hunk 1]
```diff
@@ -556,5 +556,35 @@ internal static void Arrange(this IView view, View platformView)
 
 			return null;
 		}
+
+		internal static Point? GetLocationOnScreen(this View view)
```
Comment (VladislavAntonyuk) [inline src/Core/src/Platform/Android/ViewExtensions.cs:560]: It cannot be null

File: src/Core/src/Platform/Android/ViewExtensions.cs [hunk 2]
```diff
@@ -556,5 +556,35 @@ internal static void Arrange(this IView view, View platformView)
...
+
+		internal static Point? GetLocationOnScreen(this IElement element)
+		{
+			if (element.Handler?.MauiContext == null)
+				return null;
+
+			return (element.ToPlatform())?.GetLocationOnScreen();
+		}
+
+		internal static Point? GetLocationOnScreenPx(this View view)
```
Comment (VladislavAntonyuk) [inline src/Core/src/Platform/Android/ViewExtensions.cs:575]: Output point cannot be null

File: src/Core/src/Platform/Android/ViewExtensions.cs (hunk 1 above)
Comment (PureWeen) [reply src/Core/src/Platform/Android/ViewExtensions.cs]: 👍

File: src/Core/src/Platform/Android/ViewExtensions.cs (hunk 2 above)
Comment (PureWeen) [reply src/Core/src/Platform/Android/ViewExtensions.cs]: 👍

**Review by Redth — APPROVED** [review]

**Pull request merged**